presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaConsumerManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.kafka;
+
+import com.facebook.presto.spi.HostAddress;
+import io.airlift.log.Logger;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteBufferDeserializer;
+
+import javax.inject.Inject;
+
+import java.nio.ByteBuffer;
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Manages connections to the Kafka nodes. A worker may connect to multiple Kafka nodes depending on partitions
+ * it needs to process.
+ */
+public class KafkaConsumerManager
+{
+    private static final Logger log = Logger.get(KafkaConsumerManager.class);
+    private final int maxPartitionFetchBytes;
+    private final int maxPollRecords;
+
+    @Inject
+    public KafkaConsumerManager(KafkaConnectorConfig kafkaConnectorConfig)
+    {
+        requireNonNull(kafkaConnectorConfig, "kafkaConfig is null");
+        this.maxPartitionFetchBytes = kafkaConnectorConfig.getMaxPartitionFetchBytes();
+        this.maxPollRecords = kafkaConnectorConfig.getMaxPollRecords();
+    }
+
+    KafkaConsumer<ByteBuffer, ByteBuffer> createConsumer(Long threadId, HostAddress hostAddress)
+    {
+        final Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, hostAddress.toString());
+
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, threadId.toString());
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteBufferDeserializer.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteBufferDeserializer.class.getName());
+        props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, Integer.toString(maxPollRecords));
+        props.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, maxPartitionFetchBytes);
+        props.put(ConsumerConfig.CLIENT_ID_CONFIG, String.format("%d-%s", threadId, hostAddress.toString()));
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        props.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, "org.apache.kafka.clients.consumer.RoundRobinAssignor");
+
+        Thread.currentThread().setContextClassLoader(null);
```

Comment:
why set classloader to null?

Reply:
Fixed. Was there to get ride of some exception during experiments.